hdl/WriteOnceReg.bsv

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 Oxide Computer Company
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+package WriteOnceReg;
+
+//
+// `mkWriteOnceReg` is a thin wrapper around a `Reg#(t)`, allowing it to be
+// written only once. Subsequent writes to the register are silently ignored.
+// This can for example be used to implement write once soft fuses.
+//
+module mkWriteOnceReg #(module#(Reg#(t)) m) (Reg#(t));
+    (* hide *) Reg#(t) _r <- m;
+    Reg#(Bool) written <- mkReg(False);
+
+    method t _read = _r._read;
+    method Action _write(t val);
+        if (!written) begin
+            _r <= val;
+            written <= True;
+        end
+    endmethod
+endmodule
+
+//
+// `mkWriteOnceRegA` is a thin wrapper around a `Reg#(t)`, allowing it to be
+// written only once. Subsequent writes to the register are silently ignored.
+// This can for example be used to implement write once soft fuses.
```

Comment:
Is it worth mentioning the asynchronous reset behavior of the underlying `RegA` in this comment? I don't think we _have_ to, but given we are copy/pasting the "docstring" it feels like we could.

Reply:
Yes, let me add something on this, because it may not be obvious that you could pass it a registers which is reset synchronously. Come to think of this I should probably add some typedefs, so you can create the right thing on one go.